### PR TITLE
Feature/cloudhub 183

### PR DESCRIPTION
**Pull request by MarcoInnovations**

### CODE REVIEW

File: docs/EKS_README.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# Preparing an EKS cluster in both Transit and Application accounts:
+
+## Prerequirements (*very* specific for EKS):
+* Both accounts need to adhere to the 'organizations' account structure
+* a transit VPC with three AZ's (and three private subnets) in transit (3 is recommended, 1 is bare minimum)
+* an advanced VPC with six AZ's; with six subnets: 3x private and 3x public (3 is bare minimum)
+* Each subnet needs at least to be a /26 
+* an accessible and functional proxy in the transit account to provide internet access
+* Access to a PRIVATE repository system (e.g. ATC Bitbucket) to host your own environment-configuration specific for your environment
```
Comment (greermichael) [inline docs/EKS_README.md]: Remove mention of ATC

File: docs/EKS_README.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+
+## Known Limitations:  
+
+* There are limited amount of IP's available in Transit accounts.  This may affect the creation/scaling of cluster, loadbalancers etc.  Therefore limited amount of infrastructure should be deployed in transit, with minimal resource reservations (e.g. when creating an EKS cluster with m4.large instances, some 18 IP addresses per instance are automatically reserved).  If there is an IP shortage, your cluster may not come on-line or will not scale up.).
+* Each EKS cluster takes about 12 minutes to build in either account.  This is an Amazon restriction.  Please keep that in mind when starting a build.
+* AWS resources are currently unable to resolve ATC bitbucket servers by name, please use xxx instead of yyy
```
Comment (greermichael) [inline docs/EKS_README.md]: Remove ATC, maybe use the term "on-premise"

File: docs/operation_terraform.tfvars.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+<br></br>
+http\_proxy = "[Your Proxy URL]"
+> Proxy information
+
+<br></br>
+http\_proxy\_port = 8080
+> proxy port for your environment
+
+<br></br>
+no\_proxy = "localhost,127.0.0.1,10.0.0.0/8,172.16.0.0/12,192.168.0.0/16,.elb.us-east-1.amazonaws.com,.elb.amazonaws.com,169.254.169.254,.internal,.local,.aws.cloud.bmw"
```
Comment (greermichael) [inline docs/operation_terraform.tfvars.md]: remove aws.cloud.bmw

File: docs/operation_terraform.tfvars.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+> Destinations bypassing the Proxy
+
+<br></br>
+jenkins\_config\_repo\_url = "[your git repo]"  
+> The URL to your cloned private repo for the environment;  
+> e.g.: ssh:/git@git.bmwgroup.net:7999/[q#]/[git repo name].git;
```
Comment (greermichael) [inline docs/operation_terraform.tfvars.md]: remove mention of bmwgroup.net, maybe use "internal repo git url and port"

**Comment by MarcoInnovations** [comment]
Changes proposed by greermichael adopted.

File: docs/EKS_README.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+	cd ~
+	mkdir test  
+	cd test
+	git clone https://github.com/kentrikos/aws-bootstrap.git
+	git clone https://github.com/kentrikos/template-environment-configuration.git
+	```
+
+3. Go to the cloned template-environment-configuration repository and perform the following tasks:  
+
+	* for both application and operation, create your own 'region' (e.g. us-east-1).
```
Comment (mietlinski-nc) [inline docs/EKS_README.md]:
```suggestion
	* for both application and operation, create your own 'region' folder (e.g. us-east-1).  
```

File: docs/EKS_README.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+		> copy and paste the provided eu-central-1 folder and rename it for your own region.  Do this for both Application and Operation folders.  
+
+	* modify the following files:  
+		* application/$REGION/terraform.tfvars ([example found here](application_terraform.tfvars.md))   
+		* operations/$REGION/terraform.tfvars([example found here](operation_terraform.tfvars.md))  
+		* operations/$REGION/env-eks/jenkins/parameters.yaml (config for Jenkins installation)
+		* operations/$REGION/env-eks/grafana/parameters.yaml (config for grafana installation)
+
+4. Login to your private Bitbucket
+5. create a new repo (and copy the URL to this repo)
```
Comment (mietlinski-nc) [inline docs/EKS_README.md]:
```suggestion
5. Create a new repo (and copy the URL to this repo)
```

File: docs/EKS_README.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+6. upload the template-environment-configuration to your new Bitbucket repo, making sure the folder structure stays intact.
```
Comment (mietlinski-nc) [inline docs/EKS_README.md]:
```suggestion
6. Upload the template-environment-configuration to your new Bitbucket repo, making sure the folder structure stays intact.
```

File: docs/EKS_README.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+	* URL to jenkins instance (running on port 80)
+15. In the same browser, switch to the Advanced account
+16. Create a new stack and choose to upload a template. Click on the "Choose File" button and brwose to the cloned aws-bootstrap on your local machine.  Navigate to: ~/test/aws-bootstrap/cfn/application-account
+17. Fill out the questions to fit your infrastructure
+18. Click OK / Next (no tags or other info is needed)
+19. Acknowledge that additional resources are created and click OK
+    Now resources will be created on behalve of you.  When clicking 'refresh' you can stay informed of the progress.  When the process is successfully completed, you will have:
+	* 'crossaccount role' in application account (OK: nothing much to show for here, as this account is created in the back ground)
+20. Retrieve the Jenkins URL from step 14, and login to jenkins.  
+21. Modify SSH key (imported from step zzz) to include a carriage return at the very end of the certificate
```
Comment (mietlinski-nc) [inline docs/EKS_README.md]:
```suggestion
21. Modify SSH key (imported from step 7) to include a carriage return at the very end of the certificate 
```

File: docs/operation_terraform.tfvars.md [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+
+<br></br>
+no\_proxy = "localhost,127.0.0.1,10.0.0.0/8,172.16.0.0/12,192.168.0.0/16,.elb.us-east-1.amazonaws.com,.elb.amazonaws.com,169.254.169.254,.internal,.local"
+> Destinations bypassing the Proxy
+
+<br></br>
+jenkins\_config\_repo\_url = "[your git repo]"  
+> The URL to your cloned private repo for the environment;  
+> e.g.: ssh:/git@git.bmwgroup.net:7999/[q#]/[git repo name].git;  
+> bmwgroup.net may need to be substitutes by xxx.xxx.xxx.xxx.
```
Comment (mietlinski-nc) [inline docs/operation_terraform.tfvars.md]: Please remove bmwgroup.net or the whole line.

File: docs/EKS_README.md [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+11. Fill out the questions to fit your infrastructure
+12. Click OK / Next 
+    > Recommended: Add a tag named "RunTill" with a value in format YYYY-MM-DD (This allows the EC2 resource to run uninterupted by cleanup-bots till the set date)
+13. Acknowledge that additional AWS resources are created and click OK
+14. Now resources will be created on behalve of you.  When clicking 'refresh' you can stay informed of the progress.  When the process is successfully completed, you will have:
+	* running bastion ec2 instance. 
+	* running jenkins ec2 instance (may take up to 5 minutes to appear in the EC2 list, after the bastion host is created)
+	* URL to jenkins instance (running on port 80)
+15. In the same browser, switch to the Advanced account
+16. Create a new stack and choose to upload a template. Click on the "Choose File" button and brwose to the cloned aws-bootstrap on your local machine.  Navigate to: ~/test/aws-bootstrap/cfn/application-account
```
Comment (mietlinski-nc) [inline docs/EKS_README.md]: browse

File: docs/application_terraform.tfvars.md [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+> leave this set to 'true';  
+> needed for backwards compatibility with KOPS.
+
+<br></br>
+auto\_IAM\_path = "/"
+> do not change this;  
+> needed for backwards compatibility with KOPS.
+
+<br></br>
+logs\_not\_resource = ["arn:aws:logs:*:*:ccc-*"]
```
Comment (mietlinski-nc) [inline docs/application_terraform.tfvars.md]: "arn:aws:logs:*:*:ccc-*" -> This is also customer specific

**Pull request merged**